substrate/frame/staking/src/asset.rs

```diff
@@ -18,9 +18,18 @@
 //! Contains all the interactions with [`Config::Currency`] to manipulate the underlying staking
 //! asset.
 
-use frame_support::traits::{Currency, InspectLockableCurrency, LockableCurrency};
-
-use crate::{BalanceOf, Config, NegativeImbalanceOf, PositiveImbalanceOf};
+use frame_support::traits::{
+	fungible::{
+		hold::{Balanced as FunHoldBalanced, Inspect as FunHoldInspect, Mutate as FunHoldMutate},
+		Balanced, Inspect as FunInspect,
+	},
+	tokens::Precision,
+};
+use sp_runtime::{traits::Zero, DispatchResult, Saturating};
+
```

Comment:
It is generally suggested to not have newlines between your `use` statements so that `rustfmt` would sort them.

Reply:
this is my editor